internal-api/src/main/java/datadog/trace/bootstrap/config/provider/ConfigProvider.java

```diff
@@ -255,7 +255,13 @@ public Map<String, String> getMergedMap(String key, String... aliases) {
       String value = sources[i].get(key, aliases);
       Map<String, String> parsedMap = ConfigConverter.parseMap(value, key);
       if (!parsedMap.isEmpty()) {
-        origin = sources[i].origin();
+        if (origin != ConfigOrigin.DEFAULT) {
+          // If we have already found a non-default origin, the value is calculated from multiple
+          // sources.
```

Comment:
this might keep the comment on one line...
```suggestion
          // if we already have a non-default origin, the value is calculated from multiple sources
```